server_utils/origin.go

```diff
@@ -758,11 +784,29 @@ func CheckOriginSentinelLocations(exports []OriginExport) (ok bool, err error) {
 			if path.Base(sentinelPath) != sentinelPath {
 				return false, errors.Errorf("invalid SentinelLocation path for StoragePrefix %s, file must not contain a directory. Got %s", export.StoragePrefix, export.SentinelLocation)
 			}
-			fullPath := filepath.Join(export.StoragePrefix, sentinelPath)
-			_, err := os.Stat(fullPath)
+
+			fullPath := filepath.Join(export.FederationPrefix, sentinelPath)
+			tkn, err := generateFileTestScitoken(sentinelPath)
+			if err != nil {
+				return false, errors.Wrap(err, "Failed to generate self-auth token for sentinel file check")
+			}
+
+			sentinelUrl := fmt.Sprintf("%v%v", param.Origin_Url.GetString(), fullPath)
+			req, err := http.NewRequest(http.MethodGet, sentinelUrl, nil)
+			if err != nil {
+				return false, errors.Wrap(err, "Failed to create GET request for sentinel file check")
+			}
+			req.Header.Set("Authorization", "Bearer "+tkn)
+
+			client := http.Client{Transport: config.GetTransport()}
+			resp, err := client.Do(req)
 			if err != nil {
 				return false, errors.Wrapf(err, "fail to open SentinelLocation %s for StoragePrefix %s. Collection check failed", export.SentinelLocation, export.StoragePrefix)
```

Comment:
I know you didn't write this error message, but would you mind changing it to:
"failed to open sentinel object %s for federation prefix %s."

I don't think the "collection check failed" portion of the message actually means anything, since this is an object and not a collection. I also think it makes more sense to speak in terms of the federation prefix and not the storage prefix (but you could probably change my mind on this if you feel differently).